eng/scripts/Language-Settings.ps1

```diff
@@ -141,6 +141,33 @@ function Get-java-PackageInfoFromPackageFile ($pkg, $workingDirectory)
   }
 }
 
+# Defined in common.ps1 as:
+# $GetDocsMsDevLanguageSpecificPackageInfoFn = "Get-${Language}-DocsMsDevLanguageSpecificPackageInfo"
+function Get-java-DocsMsDevLanguageSpecificPackageInfo($packageInfo, $packageSourceOverride) {
+    # If the default namespace isn't in the package info then it needs to be added
+  # Can't check if (!$packageInfo.Namespaces) in strict mode because Namespaces won't exist
+  # at all.
+  if (!($packageInfo | Get-Member Namespaces)) {
+    $version = $packageInfo.Version
+    # If the dev version is set, use that
+    if ($packageInfo.DevVersion) {
+      $version = $packageInfo.DevVersion
+    }
+    $namespaces = Fetch-Namespaces-From-Javadoc $packageInfo.Name $packageInfo.Group $version
+    if ($namespaces.Count -gt 0) {
+      $packageInfo | Add-Member -Type NoteProperty -Name "Namespaces" -Value $namespaces
+    } else {
+      # If nothing else, the package name will have "-" instead of "." which should make it evident
+      # that we couldn't find namespaces.
```

Comment:
On the second reading, the presence of `-` in a child in reference-unified.yml file would not in itself be an obvious sign that namespace fetching failed to someone who wasn't as familiar with this code.

Reply:
> On the second reading, the presence of `-` in a child in reference-unified.yml file would not in itself be an obvious sign that namespace fetching failed to someone who wasn't as familiar with this code.

Who do you think would be debugging this? Who do you think the docs team would complain to? The answer is you and I.